pkg/things/thing.go

```diff
@@ -36,6 +39,17 @@ var (
 	ErrUnauthorised = errors.New("unauthorised")
 )
 
+type commandFormat uint8
+
+const (
+	commandFormatJOSE commandFormat = iota
+	commandFormatJSON
+)
+
+type commandRequest struct {
```

Comment:
Seeing as we no longer use the `command` pattern and therefore all request to AM will use this implementation, could we rename this to something like `authenticatedRequest`? Although if you move the token adding to coapclient then I don't think you'll need this anymore.

Reply:
Renamed and moved into the coap only code. We now wrap the original payload in a new struct instead of using embedding.